### DISCUSSION TITLE
Make download file save as SiteTitle

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5562] [Ideas]
$:/core/ui/Buttons/save-wiki  contains:

```
<$button ... >
<$wikify name="site-title" text={{$:/config/SaveWikiButton/Filename}}>
<$action-sendmessage $message="tm-save-wiki" $param={{$:/config/SaveWikiButton/Template}} filename=<<site-title>>/>
</$wikify>
...
```

...but `$:/config/SaveWikiButton/Filename` is missing which I guess makes the saver fall back to save it as `tiddlywiki.html`

I wish it would first fall back on `$:/SiteTitle`.

**Comment by Jermolene** [comment]
Hi @twMat as the [docs say](https://tiddlywiki.com/#Hidden%20Setting%3A%20Filename%20for%20Save%20Wiki%20Button), if you want to use the site title as the save filename then you need to create a tiddler called `$:/config/SaveWikiButton/Filename` with the content `{{$:/SiteTitle}}.html`. That functionality was added in v5.1.19

I think the reason we didn't make it the default is because very often the site title will not be of a format suitable for use as a filename.

**Reply by twMat** [reply]
Aha, I had missed or forgotten about that setting. I would assume few people are aware of it.

>I think the reason we didn't make it the default is because very often the site title will not be of a format suitable for use as a filename.

With the recent slugify op, maybe it is now possible?

**Reply by Jermolene** [reply]
> With the recent slugify op, maybe it is now possible?

It helps, but perhaps the fundamental problem is that filenames are conventionally short for convenience, while web page titles are often several words.

Maybe we should add a settings UI for the download saver to make things clear.

**Reply by twMat** [reply]
>Maybe we should add a settings UI for the download saver to make things clear.

Yes, that makes sense. It is probably often more frequently used than several of the other accessible settings.

Could have the slugified title as default.
